tests/conftest.py

```diff
@@ -1578,16 +1582,110 @@ def rgw_obj_fixture(request):
 
 
 @pytest.fixture()
-def mcg_obj(request):
-    return mcg_obj_fixture(request)
+def mcgcli_pod(request):
+    return mcgcli_pod_fixture(request)
+
+
+@pytest.fixture(scope='session')
+def mcgcli_pod_session(request):
+    return mcgcli_pod_fixture(request)
+
+
+def mcgcli_pod_fixture(request):
+    """
+    Grab the MCG CLI tool from the NooBaa operator pod,
+    create a dedicated pod for it to allow it to run regardless of
+    the code runner's OS (since the binary does not support macOS),
+    and upload the binary to it.
+
+    """
+    nb_operator_pod = Pod(
+        **get_pods_having_label(
+            constants.NOOBAA_OPERATOR_POD_LABEL, config.ENV_DATA['cluster_namespace']
+        )[0]
+    )
+
+    def _compare_cli_hashes():
+        """
+        Verify that the remote and local CLI binaries are the same
+        in order to make sure the local bin is up to date
+
+        Returns:
+            bool: Whether the local and remote hashes are identical
+
+        """
+        remote_cli_bin_md5 = cal_md5sum(
+            nb_operator_pod,
+            constants.NOOBAA_OPERATOR_POD_CLI_PATH
+        )
+        local_cli_bin_md5 = helpers.calc_local_file_md5_sum(
+            constants.NOOBAA_OPERATOR_LOCAL_CLI_PATH
+        )
+        return remote_cli_bin_md5 == local_cli_bin_md5
+
+    mcgcli_sa = helpers.create_resource(
+        **templating.load_yaml(constants.MCGCLI_SERVICEACCOUNT_YAML)
+    )
+
+    mcg_sa_crb = helpers.create_resource(
+        **templating.load_yaml(constants.MCGCLI_SERVICEACCOUNT_CLUSTERROLEBINDING)
+    )
+
+    mcgcli_pod = helpers.create_pod(
+        namespace=config.ENV_DATA['cluster_namespace'],
+        pod_dict_path=constants.MCGCLI_POD,
+        pod_name=constants.MCGCLI_POD_NAME
+    )
+
+    if (
+        not os.path.isfile(constants.NOOBAA_OPERATOR_LOCAL_CLI_PATH)
+        or not _compare_cli_hashes()
+    ):
+        log.info('Downloading the NooBaa CLI tool from the operator pod, this might take a while.')
+        cmd = (
+            f"oc exec -n {config.ENV_DATA['cluster_namespace']} {nb_operator_pod.name}"
+            f" -- cat {constants.NOOBAA_OPERATOR_POD_CLI_PATH}"
+            f"> {constants.NOOBAA_OPERATOR_LOCAL_CLI_PATH}"
+        )
+        subprocess.run(cmd, shell=True)
+        # Make sure the binary was copied properly and has the correct permissions
+        assert os.path.isfile(constants.NOOBAA_OPERATOR_LOCAL_CLI_PATH), (
```

Comment:
should this assert and the one below be outside if statement?

Reply:
Those conditions are the same ones that are required to enter the this logic (see lines 1641-2). If they are true, the assertions will just re-check what's already checked. So I think that's a better place :)

Reply:
the 'not' part is still confusing me, 1641-42 are entering using 'not' logic ?

Reply:
specifically this case, which I think assert will be missed

```python
>>> (not True) or (not True)
False
```